Plugins/NativeXr/Source/NativeXr.cpp

```diff
@@ -2003,19 +2042,38 @@ namespace Babylon
 
                     current.insert(inputSource.ID);
 
-                    auto found = m_idToInputSource.find(inputSource.ID);
-                    if (found == m_idToInputSource.end())
+                    auto inputSourceFound = m_idToInputSource.find(inputSource.ID);
+                    if (inputSourceFound == m_idToInputSource.end())
                     {
                         // Create the new input source, which will have the correct spaces associated with it.
                         m_idToInputSource.insert({inputSource.ID, CreateXRInputSource(inputSource, env)});
+                        
+                        //Now that input Source is created, create a gamepad object if enabled for the input source
+                        inputSourceFound = m_idToInputSource.find(inputSource.ID);
+                        if (inputSource.GamepadEnabledThisFrame)
+                        {
+                            auto inputSourceVal = inputSourceFound->second.Value();
+                            m_idToGamepadObject.insert({inputSource.ID, CreateXRGamepadObject(inputSourceVal, inputSource)});
```

Comment:
After this, isn't the result of CreateXRGamepadObject stored in both `m_idToGamepadObject[inputSource.ID]` *and* `m_idToInputSource[inputSource.ID].gamepad`? Do we need it stored in two different places if it's added/updated/removed from both at the same time?

Reply:
I don't think it is stored in both Rick, we are creating napi object corresponding to the gamepad object in CreateXRGamepadObject. And we do not create that in CreateXRInputSource. So, i dont think we need to worry.

Reply:
you have a point, but I am not sure we can extract member values of a napi object. m_idToInputSource[inputSource.ID] is a napi object reference to input source. Can we extract the gamepadvalue from it?

Reply:
It looks like you can call `m_idToInputSource[inputSource.ID].Get("gamepad")` to get the gamepad object. You can either test the returned value with a call to `.IsObject()` to see if it exists, or `m_idToInputSource[inputSource.ID].Has("gamepad")`

Reply:
(I'm basing this off of what I see in the constructors for XRRay and XRHitTestSource)

Reply:
I see, i will try and see :)